src/commands/runs/ls.ts

```diff
@@ -0,0 +1,283 @@
+		DEFAULT: 'm',
+	},
+	[TimeTypes.Month]: {
+		DEFAULT: 'M',
+	},
+	[TimeTypes.Second]: {
+		DEFAULT: 's',
+	},
+	[TimeTypes.Week]: {
+		DEFAULT: 'w',
+	},
+	[TimeTypes.Year]: {
+		DEFAULT: 'y',
+	},
+});
+
+// END of TODO
+
+const multilineTimestampFormatter = new Timestamp(`YYYY-MM-DD[\n]HH:mm:ss`);
+const tableFactory = (compact = false) => {
+	const options: Record<string, unknown> = {
+		head: ['ID', 'Status', 'Results', 'Usage', 'Start Date', 'Took', 'Build Number', 'Origin'],
+		style: {
+			head: ['cyan', 'cyan', 'cyan', 'cyan', 'cyan', 'cyan', 'cyan', 'cyan'],
+			compact,
+		},
+		colAligns: ['middle', 'middle', 'middle', 'middle', 'middle', 'middle', 'middle', 'middle'],
+	};
+
+	if (compact) {
+		options.chars = {
+			'mid': '',
+			'left-mid': '',
+			'mid-mid': '',
+			'right-mid': '',
+			middle: ' ',
+			'top-mid': '─',
+			'bottom-mid': '─',
+		};
+	}
+
+	return new Table<[string, string, string, string, string, string, string, string]>(options);
+};
+
+export class RunsLsCommand extends ApifyCommand<typeof RunsLsCommand> {
+	static override description = 'Lists all runs of the Actor.';
+
+	static override flags = {
+		actor: Flags.string({
+			description:
+				'Optional Actor ID or Name to list runs for. By default, it will use the Actor from the current directory.',
+		}),
+		offset: Flags.integer({
+			description: 'Number of runs that will be skipped.',
+			default: 0,
+		}),
+		limit: Flags.integer({
+			description: 'Number of runs that will be listed.',
+			default: 10,
+		}),
+		desc: Flags.boolean({
+			description: 'Sort runs in descending order.',
+			default: false,
+		}),
+		compact: Flags.boolean({
+			description: 'Display a compact table.',
+			default: false,
+			char: 'c',
+		}),
+	};
+
+	static override enableJsonFlag = true;
+
+	async run() {
+		const { actor, desc, limit, offset, compact, json } = this.flags;
+
+		const client = await getLoggedClientOrThrow();
+
+		// TODO: technically speaking, we don't *need* an actor id to list builds. But it makes more sense to have a table of builds for a specific actor.
+		const ctx = await resolveActorContext({ providedActorNameOrId: actor, client });
+
+		if (!ctx.valid) {
+			error({
+				message: `${ctx.reason}. Please run this command in an Actor directory, or specify the Actor ID by running this command with "--actor=<id>".`,
+			});
+
+			return;
+		}
+
+		const allRuns = await client.actor(ctx.id).runs().list({ desc, limit, offset });
+
+		if (json) {
+			return allRuns;
+		}
+
+		if (!allRuns.items.length) {
+			simpleLog({
+				message: 'There are no recent runs found for this Actor.',
+			});
+
+			return;
+		}
+
+		const table = tableFactory(compact);
+
+		const message = [
+			`${chalk.reset('Showing')} ${chalk.yellow(allRuns.items.length)} out of ${chalk.yellow(allRuns.total)} runs for Actor ${chalk.yellow(ctx.userFriendlyId)} (${chalk.gray(ctx.id)})\n`,
+		];
+
+		const datasetInfos = new Map(
+			await Promise.all(
+				allRuns.items.map(async (run) =>
```

Comment:
this is effectively N+1 problem, which is never a good thing to have in your code. let's try asking on slack if there is some way to get the data with a single API request

it's not a huge deal if there wont be too many items in the list (and we use default limit of 10, so fini'ish as long as the user doesn't try to list last 200 runs...)